src/unit-taxonomy-tags-drawer/TaxonomyTagsCollapsible.jsx

```diff
@@ -0,0 +1,116 @@
+  const [target, setTarget] = React.useState(null);
+
+  const useTaxonomyTagsData = (taxonomyId, fullPathProvided) => {
+    const taxonomyTagsData = useTaxonomyTagsDataResponse(taxonomyId, fullPathProvided);
+    const isTaxonomyTagsLoaded = useIsTaxonomyTagsDataLoaded(taxonomyId, fullPathProvided);
+    return { taxonomyTagsData, isTaxonomyTagsLoaded };
+  };
+
+  const { taxonomyTagsData, isTaxonomyTagsLoaded } = useTaxonomyTagsData(id);
+
+  return (
+    <div className="taxonomy-tags-collapsible-div">
+      <Collapsible title={name} styling="card-lg" className="taxonomy-tags-collapsible">
+        <div key={id}>
+          {
+            unitTags.map((unitTag) => {
+              let level = -1; // Starting from -1 since we increment at beginning of loop
+              return unitTag.lineage.map((lineageTag) => {
+                level += 1;
+                return (<TagBubble key={`tag-${lineageTag}`} value={lineageTag} implicit={lineageTag !== unitTag.value} level={level} />);
+              });
+            })
+          }
+        </div>
+
+        <div className="taxonomy-tags-selector-menu">
+          <Button
+            ref={setTarget}
+            variant="outline-primary"
+            onClick={open}
+          >
+            Add tags
+          </Button>
+        </div>
+        <ModalPopup
+          hasArrow
+          placement="bottom"
+          positionRef={target}
+          isOpen={isOpen}
+          onClose={close}
+        >
+          <div className="bg-white p-3 shadow">
+
+            <SelectableBox.Set
+              type="checkbox"
+              name="tags"
+              columns={1}
+              ariaLabel="taxonomy tags selection"
+              className="taxonomy-tags-selectable-box-set"
+            >
+              {isTaxonomyTagsLoaded && taxonomyTagsData && taxonomyTagsData.results.map((taxonomyTag) => (
+                <TaxonomyTagsDropDownSelector
+                  key={`selector-${taxonomyTag.value}`}
+                  taxonomyId={id}
+                  taxonomyTag={taxonomyTag}
+                  level={0}
+                  useTaxonomyTagsData={useTaxonomyTagsData}
+                />
+              ))}
+            </SelectableBox.Set>
+          </div>
+        </ModalPopup>
+
+      </Collapsible>
+      <div className="tags-count-badge-div">
+        <Badge
+          variant="light"
+          pill
+          className={classNames('tags-count-badge', {
+            'tags-count-badge-empty': unitTags.length === 0,
+          })}
+        >
+          {unitTags.length}
+        </Badge>
+      </div>
+    </div>
+  );
+};
+
+TaxonomyTagsCollapsible.propTypes = {
+  taxonomyAndTagsData: PropTypes.shape({
+    id: PropTypes.number,
+    name: PropTypes.string,
+    unitTags: PropTypes.arrayOf(PropTypes.shape({
+      value: PropTypes.string,
+      lineage: PropTypes.arrayOf(PropTypes.string),
+    })),
+  }).isRequired,
+};
+
+export default injectIntl(TaxonomyTagsCollapsible);
```

Comment:
Same thing here - you're not actually using `intl` so you don't need `injectIntl`. And if you did need `intl`, it's better to use the `useIntl()` hook.